CLI/src/CLI.cs

```diff
@@ -109,36 +109,54 @@ class CLI
-                {
-                    //Delegate Event to handle diagnostics generated while loading dependencies/intrinsics
-                    Server.AddError(errorWriter, diagEvent.Path, diagEvent.Diagnostic);
-                };
-                #endregion
+                //Delegate Event to handle diagnostics generated while loading dependencies/intrinsics
+                diagDetected = true;
+                var diagnostic = diagEvent.Diagnostic;
+                Server.AddError(errorWriter, MessageCode.IntrinsicLoading,
+                    diagnostic.ColumnStart, diagnostic.ColumnEnd, diagnostic.Line,
+                    "Error while parsing " + diagEvent.Path + ": " + diagnostic, diagEvent.Path);
+            };
+            EventHandler<Tools.APIHelpers.DiagnosticsErrorEvent> DependencyErrorEvent = delegate (object sender, Tools.APIHelpers.DiagnosticsErrorEvent diagEvent)
+            {
+                //Delegate Event to handle diagnostics generated while loading dependencies/intrinsics
+                Server.AddError(errorWriter, diagEvent.Path, diagEvent.Diagnostic);
+            };
+            #endregion
+
+            depParser.CustomSymbols = Tools.APIHelpers.Helpers.LoadIntrinsic(config.Copies, config.Format, DiagnosticsErrorEvent); //Load intrinsic
+            depParser.CustomSymbols = Tools.APIHelpers.Helpers.LoadDependencies(config.Dependencies, config.Format, depParser.CustomSymbols, config.InputFiles, config.CopyFolders, DependencyErrorEvent, out List<RemarksDirective.TextNameVariation> usedCopies, out List<CopyDirective> missingCopies); //Load dependencies
 
-                depParser.CustomSymbols = Tools.APIHelpers.Helpers.LoadIntrinsic(config.Copies, config.Format, DiagnosticsErrorEvent); //Load intrinsic
-                depParser.CustomSymbols = Tools.APIHelpers.Helpers.LoadDependencies(config.Dependencies, config.Format, depParser.CustomSymbols, config.InputFiles, config.CopyFolders, DependencyErrorEvent); //Load dependencies
+            if(missingCopies.Count > 0 && !string.IsNullOrEmpty(config.HaltOnMissingCopyFilePath))
+            {
+                //Collect the missing copies
+                File.WriteAllLines(config.HaltOnMissingCopyFilePath, missingCopies.Select(c => c.TextName).Distinct());
+
+                //Create extraced copies file even if copy are missing
+                CreateExtracedCopiesFile(config, usedCopies);
 
-                if (diagDetected)
-                    throw new CopyLoadingException("Diagnostics detected while parsing Intrinsic file", null, null, logged: false, needMail: false);
+                //If copies are missing, don't try to parse main input files.
+                throw new MissingCopyException("Some copy are missing", config.Dependencies.FirstOrDefault(), null, logged: false, needMail: false);
```

Comment:
Why use FirstOrDefault ? The origin of the error could be located in any of the dependency files.

Reply:
Your right, it has been changed. Now, for each dependency that misses some copys, we throw an exception with the name of the file as argument